crates/bitwarden/src/auth/password/strength.rs

```diff
@@ -0,0 +1,99 @@
+use std::collections::HashSet;
+
+use zxcvbn::zxcvbn;
+
+const GLOBAL_INPUTS: [&str; 3] = ["bitwarden", "BW", "Bitwarden"];
+
+pub(crate) fn password_strength(
+    password: String,
+    email: String,
+    additional_inputs: Vec<String>,
+) -> u8 {
+    let email_parts = email_to_user_inputs(&email);
+    let email_ref: Vec<_> = email_parts.iter().map(String::as_str).collect();
+
+    let mut additional_inputs: HashSet<_> = additional_inputs.iter().map(String::as_str).collect();
+    additional_inputs.extend(&GLOBAL_INPUTS);
+    additional_inputs.extend(&email_ref);
+    let arr: Vec<_> = additional_inputs.into_iter().collect();
+
+    let estimate = zxcvbn(&password, &arr);
+
+    match estimate {
+        Ok(estimate) => estimate.score(),
+        _ => 0,
+    }
+}
+
+fn email_to_user_inputs(email: &str) -> Vec<String> {
+    if email.is_empty() {
```

Comment:
The empty case should already be handled by the `split_once`, as it would return none, I don't think we need to do it explicitly.